module/Finna/src/Finna/RecordDriver/SolrLrmi.php

```diff
@@ -33,9 +33,413 @@
  * @category VuFind
  * @package  RecordDrivers
  * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @author   Jaro Ravila <jaro.ravila@helsinki.fi>
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
  * @link     http://vufind.org/wiki/vufind2:record_drivers Wiki
  */
 class SolrLrmi extends SolrQdc
 {
+    /**
+     * File formats that are downloadable
+     *
+     * @var array
+     */
+    protected $downloadableFileFormats = [
+        'pdf', 'pptx', 'ppt', 'docx', 'mp4', 'mp3', 'html',
+        'avi', 'odt', 'rtf', 'txt', 'odp', 'png', 'jpeg', 'm4a'
+    ];
+
+    /**
+     * Usage rights map
+     *
+     * @var array
+     */
+    protected $usageRightsMap = [
+        'CCBY4.0' => 'CC BY 4.0',
+        'CCBYSA4.0' => 'CC BY-SA 4.0',
+        'CCBYND4.0' => 'CC BY-ND 4.0',
+        'CCBYNCND4.0' => 'CC BY-NC-ND 4.0',
+        'CCBYNCSA4.0' => 'CC BY-NC-SA 4.0',
+        'CCBYNC4.0' => 'CC BY-NC-SA 4.0'
+    ];
+
+    /**
+     * Base url for AOE record page
+     */
+    protected $aoeUrl = 'https://aoe.fi/#/materiaali/';
+
+    /**
+     * Return type of access restriction for the record.
+     *
+     * @param string $language Language
+     *
+     * @return mixed array with keys:
+     *   'copyright'   Copyright (e.g. 'CC BY 4.0')
+     *   'link'        Link to copyright info, see IndexRecord::getRightsLink
+     *   or false if no access restriction type is defined.
+     */
+    public function getAccessRestrictionsType($language = 'fi')
```

Comment:
Jos ei tarvita muita kuin nykyinen käyttöliittymän kieli, niin tehdään kuten SolrForward-ajurissa, jolloin ei tarvitse välittää toivottua kieltä:

`list($locale) = explode('-', $this->getTranslatorLocale());`

Sama kaikissa muissa metodeissa, jotka ottavat kielikoodin.